dds/DCPS/transport/framework/ReceivedDataSample.inl

```diff
@@ -41,6 +43,9 @@ ACE_INLINE
 ReceivedDataSample::~ReceivedDataSample()
 {
   DBG_ENTRY_LVL("ReceivedDataSample", "~ReceivedDataSample", 6);
+  if (!owns_sample_) {
+    sample_.release();
```

Comment:
This was confusing when I first read it because `unique_ptr::release` and `ACE_Message_Block::release` are different things with the same name.  Certainly not your issue but maybe it could use a comment.